dbt/compilation.py

```diff
@@ -117,6 +120,7 @@ def compile_node(self, node, manifest):
 
         context = dbt.context.runtime.generate(
```

Comment:
couldn't the runtime context generate the list of schemas from the list of nodes? is there a reason to do this in on-run-end hooks based on the list of results, rather than here from the list of nodes?

Reply:
If you run with models selected, this way we only get the actually executed models.